packages/autocomplete-js/src/autocomplete.ts

```diff
@@ -37,8 +43,23 @@ export function autocomplete<TItem extends BaseItem>(
         onStateChangeRef.current?.(options as any);
         props.value.core.onStateChange?.(options as any);
       },
+      shouldPanelShow:
+        optionsRef.current.shouldPanelShow ||
+        (({ state }) => {
+          const hasItems = getItemsCount(state) > 0;
+
+          return (
+            (!hasItems &&
+              Boolean(
+                hasEmptySourceTemplateRef.current ||
+                  props.value.renderer.renderEmpty
+              )) ||
```

Comment:
Perhaps store this in a `shouldRenderEmpty` variable to better understand this line.

Reply:
That's what I had in mind but it didn't really changed the readability, WDYT of this implementation? https://github.com/algolia/autocomplete.js/pull/395/commits/95298cfa50578faf4ffc173a57e7bb905bed102f

Reply:
Yep that's good!